diff -uNr pypy3.7-v7.3.7-src/lib-python/3/distutils/command/build.py pypy3.7-v7.3.7-src.mod/lib-python/3/distutils/command/build.py
--- pypy3.7-v7.3.7-src/lib-python/3/distutils/command/build.py          2021-10-24 22:07:11.000000000 +0800
+++ pypy3.7-v7.3.7-src.mod/lib-python/3/distutils/command/build.py      2022-01-17 22:59:13.377190200 +0800
@@ -119,6 +119,7 @@
         if self.executable is None and sys.executable:
             self.executable = os.path.normpath(sys.executable)

+        self.parallel = 1
         if isinstance(self.parallel, str):
             try:
                 self.parallel = int(self.parallel)
