diff -uNr pypy3.7-v7.3.7-src/pypy/module/imp/importing.py pypy3.7-v7.3.7-src.mod/pypy/module/imp/importing.py
--- pypy3.7-v7.3.7-src/pypy/module/imp/importing.py        2021-10-24 22:07:11.000000000 +0800
+++ pypy3.7-v7.3.7-src.mod/pypy/module/imp/importing.py    2022-02-08 16:41:15.829099000 +0800
@@ -41,21 +41,8 @@
     if not space.config.translating:
         soabi += 'i'
 
-    platform_name = sys.platform
-    if platform_name.startswith('linux'):
-        if re.match('(i[3-6]86|x86_64)$', platform.machine()):
-            if sys.maxsize < 2**32:
-                platform_name = 'i686-linux-gnu'
-                # xxx should detect if we are inside 'x32', but not for now
-                # because it's not supported anyway by PyPy.  (Relying
-                # on platform.machine() does not work, it may return x86_64
-                # anyway)
-            else:
-                platform_name = 'x86_64-linux-gnu'
-        else:
-            platform_name = 'linux-gnu'
-    elif platform_name == 'win32' and sys.maxsize > 2**32:
-        platform_name = 'win_amd64'
+    # XXX: I think it should not be '(x64_64/i686-)linux-gnu' anyway, maybe 'linux-android/bionic'?
+    platform_name = "linux-android"
 
     soabi += '-' + platform_name
 
